Processors/Decoders/PowerPC/PowerPC.cpp

```diff
@@ -0,0 +1,347 @@
+//
+//  PowerPC.cpp
+//  Clock Signal
+//
+//  Created by Thomas Harte on 12/30/20.
+//  Copyright © 2020 Thomas Harte. All rights reserved.
+//
+
+#include "PowerPC.hpp"
+
+using namespace CPU::Decoder::PowerPC;
+
+Decoder::Decoder(Model model) : model_(model) {}
+
+Instruction Decoder::decode(uint32_t opcode) {
+	// Quick bluffer's guide to PowerPC instruction encoding:
+	//
+	// There is a six-bit field at the very top of the instruction.
+	// Sometimes that fully identifies an instruction, but usually
+	// it doesn't.
+	//
+	// There is an addition 9- or 10-bit field starting one bit above
+	// least significant that disambiguates the rest. Strictly speaking
+	// it's a 10-bit field, but the mnemonics for many instructions treat
+	// it as a 9-bit field with a flag at the top.
+	//
+	// I've decided to hew directly to the mnemonics.
+	//
+	// Various opcodes in the 1995 documentation define reserved bits,
+	// which are given the nominal value of 0. It does not give a formal
+	// definition of a reserved bit. As a result this code does not
+	// currently check the value of reserved bits. That may need to change
+	// if/when I add support for extended instruction sets.
+
+#define Bind(mask, operation)				case mask: return Instruction(Operation::operation, opcode);
+#define BindSupervisor(mask, operation)		case mask: return Instruction(Operation::operation, opcode, true);
+#define BindConditional(condition, mask, operation)	\
+	case mask: \
+		if(condition()) return Instruction(Operation::operation, opcode);	\
+	return Instruction(opcode);
+#define BindSupervisorConditional(condition, mask, operation)	\
+	case mask: \
+		if(condition()) return Instruction(Operation::operation, opcode, true);	\
+	return Instruction(opcode);
```

Comment:
Would the supervisor true argument be required here as well?

Reply:
No, wait, I take that back. `Instruction(opcode)` with no operation provided produces an unrecognised instruction. Which aren't subject to supervisor/user distinction.

Practically speaking, I suspect other errors lurk here — they'll become clearer if/when I actually use this code. For now it's just to ensure that I set off on the right path for these things (and PowerPC specifically since that's the book that came to hand).